app/assets/javascripts/host_edit.js

```diff
@@ -173,7 +173,7 @@ function submit_with_all_params() {
     success: function(response) {
       // workaround for redirecting to the new host details page
       if (!response.includes('id="main"')) {
-        return tfm.nav.pushUrl('/new/hosts/' + fqdn);
+        return tfm.nav.pushUrl('/new/hosts/' + host_unique_name);
```

Comment:
Please see https://github.com/theforeman/foreman/pull/9047#discussion_r790797829.

Reply:
Unfortunatelly I've already tried that. I'm adding those parameters to indicate those were tried with no success.

Reply:
As of now this fixes the issue for the two cases that I'm aware of and added tests to. It is not perfect and it is hacky, but without serious rewrite of `host_edit.js` I do not see a better way.